server.js

```diff
@@ -1,22 +1,42 @@
 import express from "express";
 import cors from "cors";
 import mongoose from "mongoose";
+import netflixData from "./data/netflix-titles.json";
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// import avocadoSalesData from "./data/avocado-sales.json";
-// import booksData from "./data/books.json";
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
 mongoose.Promise = Promise;
 
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
+const Title = mongoose.model("Title", {
+  show_id: Number,
+  title: String,
+  director: String,
+  cast: String,
+  country: String,
+  date_added: String,
+  release_year: Number,
+  rating: String,
+  duration: String,
+  listed_in: String,
+  description: String,
+  type: String
+
+})
+
+if(process.env.RESET_DB) {
+  console.log("resetting db")
+  const resetDataBase = async () => {
+    await Title.deleteMany();
+
+    netflixData.forEach(item => {
+      const newTitle = new Title(item);
+      newTitle.save();
+    })
+  }
+  resetDataBase();
+}
+
 const port = process.env.PORT || 8080;
 const app = express();
 
```

Comment:
Another thing that I learned is the below code:
This is for when you click on your backend link and if it for exemple takes some time for it to connect or not.

//an express middleware to check if mongoose is ready and throw an error if not. Note: Should always be added before the router.
app.use((req, res, next) => {
  if (mongoose.connection.readyState === 1) {
    next()
  } else {
    res.status(503).json({
      error: `Service unavailable`
    })
  }
})